HW_14/2_exercise.py

```diff
@@ -0,0 +1,37 @@
+class Writer:
+    def write(self, data):
+        raise NotImplementedError()
+
+
+class FileWriter(Writer):
+    def __init__(self, filename):
+        self.filename = filename
+
+    def write(self, data):
+        with open(self.filename, 'w') as file:
+            file.write(data)
+        print(f"Data written to file: {self.filename}")
+
+
+class WriterProxy(Writer):
+    def __init__(self, writer):
+        self.writer = writer
+
+    def write(self, data):
+        if self._is_authorized():
+            self.writer.write(data)
+        else:
+            print("Unauthorized access. Write operation not allowed.")
```

Comment:
proxy is more about working with data. Not authorization. I want you to thhink about when to write to file and when not in scope of file content